packages/react-search-ui-views/src/Facet.tsx

```diff
@@ -0,0 +1,41 @@
+import {
+  FacetValue,
+  FieldValue,
+  FilterType,
+  SearchContextState
+} from "@elastic/search-ui";
+import { BaseContainerProps } from "./types";
+
+export type FacetViewProps = {
+  className?: string;
+  label: string;
+  onMoreClick: () => void;
+  onRemove: (value: FieldValue) => void;
+  onChange: (value: FieldValue) => void;
+  onSelect: (value: FieldValue) => void;
+  options: FacetValue[];
+  showMore: boolean;
+  values: any;
+  showSearch: boolean;
+  onSearch: (value: string) => void;
+  searchPlaceholder: string;
+};
+
+export type FacetContainerContext = Pick<
+  SearchContextState,
+  | "addFilter"
+  | "removeFilter"
+  | "setFilter"
+  | "a11yNotify"
+  | "filters"
+  | "facets"
+>;
+
+export type FacetContainerProps = BaseContainerProps & {
+  filterType: FilterType;
+  show?: number;
+  view?: React.ComponentType<FacetViewProps>;
+  isFilterable: boolean;
+  field: string;
+  label: string;
+} & FacetContainerContext;
```

Comment:
Maybe we could put this in `types.ts` file? Right now it looks like one of the components.

Reply:
yeh looking at it again il move it in types. thanks!